src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -5,25 +5,80 @@
 
 using System;
 using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
 using Microsoft.Health.Fhir.Core.Features.Operations;
 using Microsoft.Health.Fhir.Core.Features.Operations.Export.Models;
 using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.SqlServer.Configs;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+using Microsoft.IO;
+using Newtonsoft.Json;
 
 namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
 {
     public class SqlServerFhirOperationDataStore : IFhirOperationDataStore
     {
+        internal static readonly Encoding ResourceEncoding = new UnicodeEncoding(bigEndian: false, byteOrderMark: false);
+
+        private readonly SqlServerDataStoreConfiguration _configuration;
+        private readonly ILogger<SqlServerFhirOperationDataStore> _logger;
+        private readonly RecyclableMemoryStreamManager _memoryStreamManager;
+
+        public SqlServerFhirOperationDataStore(SqlServerDataStoreConfiguration configuration, ILogger<SqlServerFhirOperationDataStore> logger)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _configuration = configuration;
+            _logger = logger;
+            _memoryStreamManager = new RecyclableMemoryStreamManager();
+        }
+
         public Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(ushort maximumNumberOfConcurrentJobsAllowed, TimeSpan jobHeartbeatTimeoutThreshold, CancellationToken cancellationToken)
         {
             IReadOnlyCollection<ExportJobOutcome> returnValue = new List<ExportJobOutcome>();
             return Task.FromResult(returnValue);
         }
 
-        public Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken)
+        public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            using (var connection = new SqlConnection(_configuration.ConnectionString))
```

Comment:
Not sure if the export can ever be part of a FHIR transaction, but if so the connection will need to be handled differently.

Reply:
@namadabu @jackliums what are your thoughts on having export available as part of a FHIR transaction?

Reply:
The spec specific calls out that operations are supported within batch/transaction so I suppose that it should be allowed. But it does make reference detection harder though. Should ask the FHIR community.

Reply:
Created a post [here](http://community.fhir.org/t/bulk-export-and-fhir-transactions/1688) and [#AB71716](https://microsofthealth.visualstudio.com/Health/_workitems/edit/71716) to look into this further.